wgsl/index.bs

```diff
@@ -9360,6 +9360,44 @@ The following algorithm describes how to compute these tags for a given function
 
 Note: The entire graph can be destroyed at this point. The tags listed above are all that we need to remember to analyze callers of this function.
 
+### Pointer Desugaring ### {#pointer-desguar}
+
+Each [=formal parameter|parameter=] of [=pointer type=] in the [=address
+spaces/function=] address space is desugared as a local variable declaration
+whose initial value is equivalent to dereferencing the parameter.
+That is, function address space pointers are viewed as aliases to a local
+variable declaration.
+
+Each [=let-declaration=] with an [=effective-value-type=] that is a [=pointer
+type=] is desugared as follows:
+* The initializer expression of the declaration is recorded.
+* Each [=identifier=] that [=resolves=] is substituted with the recorded
```

Comment:
Is there a technical problem here? Because the indexing into those desugared pointers could have its uniformity properties changed.  They may refer to variables that have had full assignments on them in the meantime, or vice versa.   Basically, you have to properly capture the values of variables at the point of recording the initializer for the let-decls.

E.g. see how 'foo' evolves through desugaring.  Yet, variable i receives a full assignment that makes it uniform halfway through the function.
```rust
@group(0) @binding(0) var t: texture_2d<f32>;
@group(0) @binding(1) var s: sampler;

fn foo(parami:i32) -> vec4<f32> {
   var i = parami;
   var v = array<i32,2>(-1,1);
   let p = &v[i];
   if (v[i] > 0) {
      return textureSample(t,s,vec2(.0));  // causes uniformity error.
   }
   i = 0;   // full assignment
   if (v[i] > 0) {
      return textureSample(t,s,vec2(.0)); // no uniformity error
   }
   return vec4<f32>();
}

fn foo_via_pointers(parami:i32) -> vec4<f32> {
   var i = parami;
   var v = array<i32,2>(-1,1);
   let p = &v[i];
   if (*p > 0) {
      return textureSample(t,s,vec2(.0));  // should cause uniformity error.
   }
   i = 0;  // full assignment
   if (*p > 0) {
      return textureSample(t,s,vec2(.0)); // should still cause uniformity error
   }
   return vec4<f32>();
}

fn foo_via_pointers_desugared(parami:i32) -> vec4<f32> {
   var i = parami;
   var v = array<i32,2>(-1,1);
   let p = &v[i];
   if (*(&v[i]) > 0) {
      return textureSample(t,s,vec2(.0));  // should cause uniformity error.
   }
   i = 0;  // full assignment
   if (*(&v[i]) > 0) {
      return textureSample(t,s,vec2(.0)); // should still cause uniformity error
                                          // but it won't
   }
   return vec4<f32>();
}

@fragment
fn main(@builtin(position) p: vec4<f32>) -> @location(0) vec4<f32> {
  return foo(i32(trunc(p.x)));
}
```

Reply:
Good catch. The desugaring should be capturing the values of mutable variables other than the root identifier.